File: usecase/src/main/java/br/org/restaurantepopular/usecase/category/ICategoryRepository.java
package br.org.restaurantepopular.usecase.category;

import br.org.restaurantepopular.entity.Category;

import java.util.List;
import java.util.Optional;

public interface ICategoryRepository {

    void save(Category category);

    void update(Category category);

    List<Category> listCategories();

    Optional<Category> findById(String id);

    void delete(String id);
}

File: usecase/src/main/java/br/org/restaurantepopular/usecase/category/ICategoryRepository.kt
package br.org.restaurantepopular.usecase.category

import br.org.restaurantepopular.entity.Category

interface ICategoryRepository {
    fun save(category: Category): Category
    fun update(category: Category)
    fun listCategories(): List<Category>
    fun findById(id: String): Category
    fun delete(id: String)
}